pkg/context/context.go

```diff
@@ -24,10 +24,27 @@ type Context struct {
 	alert                option.Alerter
 }
 
+// WithContext returns an option.Option that contains a pre-existing Context
+// struct. This is useful for some internal code that sets up snapshots.
+func WithContext(ctx *Context) *option.Option {
+	return &option.Option{
+		Context: ctx,
+	}
+}
+
 // New returns a Context struct pointer that has had various options set on it
 func New(opts ...*option.Option) *Context {
 	merged := option.Merge(opts...)
-	ctx := &Context{
+	var ctx *Context
+	if merged.Context != nil {
+		var castOK bool
+		ctx, castOK = merged.Context.(*Context)
+		if !castOK {
+			panic("passed in a non-Context for the WithContext() function!")
```

Comment:
I usually dislike `panic`ing into library code, but this is probably one of the very few places on which is useful. So I'm fine.

Reply:
yeah, this is more of a compile-time thing since mostly it's just checking to make sure we're setting the interface properly.